modules/pytket-quantinuum/pytket/extensions/quantinuum/backends/quantinuum.py

```diff
@@ -575,12 +604,21 @@ def cost(
                 "Circuit does not satisfy predicates of backend."
                 + " Try running `backend.get_compiled_circuit` first"
             )
-        if syntax_checker:
-            backend = QuantinuumBackend(syntax_checker, _api_handler=self._api_handler)
-        else:
-            backend = QuantinuumBackend(
-                _infer_syntax_checker(self._device_name), _api_handler=self._api_handler
+
+        try:
+            syntax_checker = (
```

Comment:
Are you sure you want to override here the variable?

Reply:
yeah, I don't see any issue with it